src/object_.d

```diff
@@ -169,8 +169,11 @@ bool opEquals(Object lhs, Object rhs)
     if (lhs is null || rhs is null) return false;
 
     // If same exact type => one call to method opEquals
-    if (typeid(lhs) is typeid(rhs) || typeid(lhs).opEquals(typeid(rhs)))
+    if (!__ctfe && // CTFE doesn't like comparing typeid
```

Comment:
> I thought it was just an optimization, saving one opEquals call.

Exactly, but that's the problem, because opEquals can have a side-effect so CTFE and runtime would disagree.